core/src/credential.rs

```diff
@@ -0,0 +1,147 @@
+
+#![allow(unused_variables)]
+#![allow(dead_code)]
+use std::{error::Error, time::{Instant, SystemTime}};
+
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+// cred_subject is a generic that implements trait X
+// trait X allows us to encode that object into JSON-LD
+// We provide types that implement trait X for the cred types that we support
+// Users can also user their own types that implement trait X if they need a different structure
+// --- 
+// Default context and Cred types are defaulted but can be redefined 
+
+type VCContext = [&'static str];
+
+pub(crate) const JSON_LD_CONTEXT:  VCContext = ["https://www.w3.org/2018/credentials/v1", "https://www.w3.org/2018/credentials/examples/v1"];
+
+pub const CRED_TYPE_PERMANENT_RESIDENT_CARD: str = "PermanentResidentCard";
```

Comment:
How about we remove the `CRED_TYPE` prefix and create a custom type called `CredentialType` or something like that?

Reply:
Agree, it can be enum based type, it is enum by nature

Reply:
Yeah as long as we also get the string format without issue then it'll work out 👍🏽

Reply:
I've tried to figure out the best possible structure for this, but const approach seems to be more suitable here for a couple of reasons:
- neither enum nor Struct is iterable which still puts on us the responsibility to maintain the array of available credentials manually, the same as with constants, so this approach doesn't really give us anything 
- both enum and struct require initialization in order for these two fields to be ready for usage, moreover there is no clear semantical definition that this is constant and can't be changed
- with const approach user sees predefined values for this library without any initialization and can use these constants in the method calls right away, like:
`ssi.creade_credential(ssi.CRED_TYPE_CARD, param1, param2)`